### PR TITLE
Add CI for Triton CPU backend

### PR DESCRIPTION
**Pull request by int3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136612
* __->__ #135342

Where possible, I have marked failing tests (which we intend to fix or triage) as `@xfail_if_triton_cpu`. This will help us track progress of the Triton CPU backend over time. Tests that I don't think we need to address, or that are flaky, have been marked as skips.

Successful CI run: https://github.com/pytorch/pytorch/actions/runs/10822238062/job/30028284549

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by int3** [comment]
Alright this finally succeeded

File: .ci/docker/common/install_triton.sh [hunk 1]
```diff
@@ -44,9 +47,10 @@ chown -R jenkins /var/lib/jenkins/triton
 chgrp -R jenkins /var/lib/jenkins/triton
 pushd /var/lib/jenkins/
 
-as_jenkins git clone ${TRITON_REPO} triton
+as_jenkins git clone --recursive ${TRITON_REPO} triton
 cd triton
 as_jenkins git checkout ${TRITON_PINNED_COMMIT}
+as_jenkins git submodule update --init --recursive
```
Comment (int3) [inline .ci/docker/common/install_triton.sh:53]: triton-cpu uses submodules

File: torch/testing/_internal/inductor_utils.py [hunk 2]
```diff
@@ -74,6 +74,7 @@ def _check_has_dynamic_shape(
 def skipDeviceIf(cond, msg, *, device):
     if cond:
         def decorate_fn(fn):
+            @functools.wraps(fn)
```
Comment (int3) [inline torch/testing/_internal/inductor_utils.py:77]: this is needed in order for the special `_expected_failure` attributes to be propagated up

**Review by jansel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

File: .ci/docker/build.sh [hunk 3]
```diff
@@ -381,6 +381,13 @@ case "$image" in
     HALIDE=yes
     TRITON=yes
     ;;
+  pytorch-linux-jammy-py3.12-triton-cpu)
+    CUDA_VERSION=12.4
```
Comment (malfet) [inline .ci/docker/build.sh:385]: You probably don't want CUDA here, do you?
```suggestion
```

**Review by malfet — APPROVED** [review]

File: .ci/docker/build.sh (hunk 3 above)
Comment (int3) [reply .ci/docker/build.sh:385]: This seems to have caused an error due to libmpi_cxx not being found. I think it got installed as part of the CUDA PyTorch install. I guess I could figure out how to install it independently, but just installing CUDA seems easier

**Pull request closed without merging**